actions/ulog_ingestion/src/ulog_ingestion/__main__.py

```diff
@@ -0,0 +1,295 @@
+    )
+
+    print(
+        f"Setting default representation for topic: {topic_name_roboto}, file_id: {dataset.get_file_info(relative_file_name).file_id}"
+    )
+
+    # Set Default Topic Representation
+    topic.set_default_representation(
+        request=topics.SetDefaultRepresentationRequest(
+            association=Association(
+                association_type=AssociationType.File,
+                association_id=dataset.get_file_info(relative_file_name).file_id,
+            ),
+            org_id=org_id,
+            storage_format=topics.RepresentationStorageFormat.MCAP,
+            version=1,
+        )
+    )
+    return
+
+
+def ingest_ulog(ulog_file_path: str, topics: List[str] = None):
+    """
+
+    This function creates topic entries, message path records, and MCAP files from a ULog file.
+
+    Args:
+    - ulog_file_path: Path to the .ulg file.
+    - messages: List of messages to process, separated by commas.
+
+    Returns:
+    - None
+    """
+    msg_filter = topics.split(",") if topics else None
+    print(ulog_file_path)
+
+    ulog = ULog(ulog_file_path, msg_filter, True)
+
+    _, input_dir, _, _, _ = setup_env()
+
+    output_dir_path_mcap, output_dir_temp = utils.setup_output_folder_structure(
+        ulog_file_path, input_dir
+    )
+    start_time = time.time()
+
+    # This is a temporary fix for Multi Information messages with the same name
+    # https://docs.px4.io/main/en/dev_log/ulog_file_format.html#m-multi-information-message
+    # Similar to Plotjuggler approach
+    # TODO: handle this in a better way
+    message_names_with_multi_id_list = list()
+    for d in sorted(ulog.data_list, key=lambda obj: obj.name):
+        print(f"topic name: {d.name}")
+        if d.multi_id > 0:
+            message_names_with_multi_id_list.append(d.name)
+
+    message_names_with_multi_id_list = list(set(message_names_with_multi_id_list))
+
+    print(f"message_names_with_multi_id_list: {message_names_with_multi_id_list}")
+
+    # Prepare Arguments for Parallel Processing
+    args_list = [
+        (
+            ulog_file_path,
+            ulog_data_object,
+            ulog.message_formats,
+            message_names_with_multi_id_list,
+            output_dir_path_mcap,
+            output_dir_temp,
+        )
+        for ulog_data_object in sorted(ulog.data_list, key=lambda obj: obj.name)
+    ]
+
+    with ProcessPoolExecutor() as executor:
+        # Submit all tasks and collect Future objects
+        futures = [executor.submit(process_data, *args) for args in args_list]
+
+        for future in as_completed(futures):
+            try:
+                result = future.result()
+                print("Task completed successfully", result)
+            except Exception as exc:
+                print(f"Task generated an exception: {exc}")
+
+    end_time = time.time()
+    elapsed_time = end_time - start_time
+    print(f"The topic create function took {elapsed_time} seconds to run.")
```

Comment:
There's an inconsistency in what this is outputting in the logs for https://app-beta.roboto.ai/invocations/iv_15ebjtdbjl4o (Roboto Public, Beta) versus how long the website is claiming this step took. The website is saying the `Processing` step took ~13.5min, while this log line is saying `"The topic create function took 43.96817708015442 seconds to run."`